hooks/react-client/useNavigationState.ts

```diff
@@ -0,0 +1,40 @@
+'use client';
+
+import type { RefObject } from 'react';
+import { useContext, useEffect } from 'react';
+
+import { NavigationStateContext } from '@/providers/navigationStateProvider';
+import { debounce } from '@/util/debounce';
+
+const useNavigationState = <T extends HTMLElement>(
+  id: string,
```

Comment:
Need alternate name suggestions for id

Reply:
Why?

Reply:
I felt that there could be a better name 😅